### PR TITLE
Use system sound play instead of IrrKlang

### PR DESCRIPTION
**Pull request by JohnThomson**
Also adds a way to test interactively
Therefore makes TestApp x86 (since SIL.Media is)

---
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/libpalaso/548)

### CODE REVIEW

File: SIL.Media/AudioIrrKlangSession.cs [hunk 1]
```diff
@@ -112,38 +111,79 @@ public bool CanPlay
...
 				{
-					// The most likely exception is that the sound has finished playing
+					// Try to clean things up as best we can...no easy way to test this, though.
+					// Even though we're about to try to play it another way, we won't know when that's done,
+					// and we don't want to be permanently in the playing state.
+					IsPlaying = false;
+					// And, in case something critical is waiting for this...
+					PlaybackStopped?.Invoke(this, new EventArgs());
+					// Maybe the system has another way of playing it that works? e.g., most default players will handle mp3.
+					Process.Start(FilePath);
```
Comment (hatton) [inline SIL.Media/AudioIrrKlangSession.cs]: I'm dubious about this, would suggest not trying some other method.

File: SIL.Media/AudioIrrKlangSession.cs [hunk 2]
```diff
@@ -112,38 +111,79 @@ public bool CanPlay
...
+		SoundPlayer _player = new SoundPlayer();
+
 		public void Play()
 		{
 			if (IsRecording)
 				throw new ApplicationException("Can't play while recording.");
-
-			if (_sound != null)
+			if (!File.Exists(FilePath))
+				throw new FileNotFoundException("No file to play");
```
Comment (hatton) [inline SIL.Media/AudioIrrKlangSession.cs]: I think the previous message was less ambiguous: "Could not find sound file"

File: TestApps/SIL.Windows.Forms.TestApp/TestAppForm.cs [hunk 3]
```diff
@@ -21,6 +22,8 @@
 using SIL.Windows.Forms.SettingProtection;
 using SIL.Windows.Forms.WritingSystems;
 using SIL.WritingSystems;
+using SIL.Media;
+using Microsoft.VisualBasic.Devices;
```
Comment (hatton) [inline TestApps/SIL.Windows.Forms.TestApp/TestAppForm.cs]: I can't figure out what of Microsoft.VisualBasic.Devices is being used. I suppose if I had the code in an IDE I would comment it out and see what turns red :-)

**Comment by andrew-polk** [comment]
Review status: 0 of 4 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[TestApps/SIL.Windows.Forms.TestApp/TestAppForm.cs, line 26 at r1](https://reviewable.io:443/reviews/sillsdev/libpalaso/548#-KnaDDsyY3adzYMvjnmB-r1-26:-KndaTPKE9zBO6AxNuhI:b-k0xooe) ([raw file](https://github.com/sillsdev/libpalaso/blob/ca02c1af284f862aaeb7ed47374654a6976caa72/TestApps/SIL.Windows.Forms.TestApp/TestAppForm.cs#L26)):*
<details><summary><i>Previously, hatton (John Hatton) wrote…</i></summary><blockquote>

I can't figure out what of Microsoft.VisualBasic.Devices is being used. I suppose if I had the code in an IDE I would comment it out and see what turns red :-)
</blockquote></details>

Good catch. It builds without it and without the added reference in the project.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/sillsdev/libpalaso/548)*

**Comment by JohnThomson** [comment]
Review status: 0 of 4 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[SIL.Media/AudioIrrKlangSession.cs, line 121 at r1](https://reviewable.io:443/reviews/sillsdev/libpalaso/548#-KnaDDsyY3adzYMvjnm8-r1-121:-KneBbWDXerF_1WqGNNj:b-896fix) ([raw file](https://github.com/sillsdev/libpalaso/blob/ca02c1af284f862aaeb7ed47374654a6976caa72/SIL.Media/AudioIrrKlangSession.cs#L121)):*
<details><summary><i>Previously, hatton (John Hatton) wrote…</i></summary><blockquote>

I think the previous message was less ambiguous: "Could not find sound file"
</blockquote></details>

Done.

---

*[SIL.Media/AudioIrrKlangSession.cs, line 154 at r1](https://reviewable.io:443/reviews/sillsdev/libpalaso/548#-KnaDDsyY3adzYMvjnm8-r1-154:-KneBd4O4tjzCtzS9x_Y:b4yclp3) ([raw file](https://github.com/sillsdev/libpalaso/blob/ca02c1af284f862aaeb7ed47374654a6976caa72/SIL.Media/AudioIrrKlangSession.cs#L154)):*
<details><summary><i>Previously, hatton (John Hatton) wrote…</i></summary><blockquote>

I'm dubious about this, would suggest not trying some other method.
</blockquote></details>

The old code did, but I'm fine with not doing so. Done.

---

*[TestApps/SIL.Windows.Forms.TestApp/TestAppForm.cs, line 26 at r1](https://reviewable.io:443/reviews/sillsdev/libpalaso/548#-KnaDDsyY3adzYMvjnmB-r1-26:-KneCKSzvjis_d_8YY0r:bvpb1jl) ([raw file](https://github.com/sillsdev/libpalaso/blob/ca02c1af284f862aaeb7ed47374654a6976caa72/TestApps/SIL.Windows.Forms.TestApp/TestAppForm.cs#L26)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

Good catch. It builds without it and without the added reference in the project.
</blockquote></details>

Was part of a failed attempt at an easier way to tell whether the mouse button was still pressed. Removed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/sillsdev/libpalaso/548)*

**Comment by andrew-polk** [comment]
Review status: 0 of 5 files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[TestApps/SIL.Windows.Forms.TestApp/SIL.Windows.Forms.TestApp.csproj, line 272 at r2](https://reviewable.io:443/reviews/sillsdev/libpalaso/548#-KneE_lC7S3fWYeTX_le:-KneE_lDm4SV3ZZbRNT9:bmarxiu) ([raw file](https://github.com/sillsdev/libpalaso/blob/14b0b04b32f3c6dd91d50e94d5a07a347f8bbb0a/TestApps/SIL.Windows.Forms.TestApp/SIL.Windows.Forms.TestApp.csproj#L272)):*
> ```XML
>       <HintPath>..\..\lib\Debug\L10NSharp.dll</HintPath>
>     </Reference>
>     <Reference Include="Microsoft.VisualBasic" />
> ```

This needs to be removed, too.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/sillsdev/libpalaso/548)*

**Comment by hatton** [comment]
Review status: 0 of 5 files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[SIL.Media/AudioIrrKlangSession.cs, line 154 at r1](https://reviewable.io:443/reviews/sillsdev/libpalaso/548#-KnaDDsyY3adzYMvjnm8-r1-154:-KneHcNLV0XZTAnA3-3u:b-wnxewb) ([raw file](https://github.com/sillsdev/libpalaso/blob/ca02c1af284f862aaeb7ed47374654a6976caa72/SIL.Media/AudioIrrKlangSession.cs#L154)):*
<details><summary><i>Previously, JohnThomson (John Thomson) wrote…</i></summary><blockquote>

The old code did, but I'm fine with not doing so. Done.
</blockquote></details>

Probably for the best. I did look at the code over there on the  left of this diff, and I don't see where it tries a different  approach. Must be somewhere else.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/sillsdev/libpalaso/548)*

**Pull request merged**